### PR TITLE
[release/v2.18] Bump Kubernetes versions to 1.20.14, 1.21.8, 1.22.5 and use Go 1.16.12

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:
Corresponding to https://github.com/kubermatic/kubermatic/pull/8472, this adds Kubernetes 1.20.14, 1.21.8 and 1.22.5 to the supported versions in KKP 2.18. 1.21.x and 1.22.x are auto-upgraded to apply security fixes.

In addition, the Go version is bumped up to 1.16.12 for CVE fixes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Add support for Kubernetes 1.22.5, 1.21.8, and 1.20.14
Automatically upgrade clusters running Kubernetes 1.21 to 1.21.8 to include fixes for CVE-2021-44716 and CVE-2021-44717
Automatically upgrade clusters running Kubernetes 1.22 to 1.22.5 to include fixes for CVE-2021-44716 and CVE-2021-44717
Update to Go 1.16.12
```

Signed-off-by: Marvin Beckers <marvin@kubermatic.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -96,6 +96,8 @@ require (
 )
 
 replace (
+	golang.org/x/net => golang.org/x/net v0.0.0-20211209124913-491a49abca63 // fixes CVE-2021-44716
```
Comment (xmudrii) [inline go.mod:99]: We don't need this. Can you please run this instead, and then drop this:

```
go get -v -d golang.org/x/net@491a49abca63
```

File: pkg/controller/seed-controller-manager/kubernetes/resources_integration_test.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build integration
```
Comment (xmudrii) [inline pkg/controller/seed-controller-manager/kubernetes/resources_integration_test.go]: This shouldn't be there, because KKP 2.18 is still using Go 1.16.

**Review by xmudrii — CHANGES_REQUESTED** [review]
Left some comments, otherwise LGTM.

File: go.mod (hunk 1 above)
Comment (embik) [reply go.mod:99]: As mentioned on Slack, I think it's somewhere down in the dependency tree as it's not in `go.mod` but in `go.sum`, and this overrides it for anyone importing it, but we don't have it in `go.mod` so I wouldn't import it via `go get` to be honest.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**